snowflake-api/src/lib.rs

```diff
@@ -225,22 +245,53 @@ impl SnowflakeApi {
             .with_token(info.creds.aws_token)
             .build()?;
 
+        let s3_mutex = Arc::new(Mutex::new(s3));
```

Comment:
Wouldn't using mutex here defeat the purpose of uploading files in parallel? Since you're going to lock IO intensive task into a single thread

Reply:
I think your right! Do you have any suggestions for a better solution? This is probably why I didn't see performance gains.

Reply:
It appears to me to be thread safe, I guess you will be ok without Mutex

Reply:
This sped up the code 5 times!